File: Examples/VBasic/GroupDocs.Text.Examples.VBasic/Tools.vb

Imports System.IO
Imports GroupDocs.Text.Detectors
Imports GroupDocs.Text.Detectors.Encoding
Imports GroupDocs.Text.Extractors.Text
Imports System.Text
Imports GroupDocs.Text

Public Class Tools
    Public Class EncodingDetection
        ''' <summary>
        ''' Detects encoding of a filestream when BOM is present
        ''' </summary>
        ''' <param name="fileName"></param>
        Public Shared Sub ExtractEncodingByBOM(fileName As String)
            'ExStart:ExtractEncodingByBOM
            Try
                Dim detector As New EncodingDetector(Encoding.GetEncoding(1251))
                'get file actual path
                Dim filePath As [String] = Common.getFilePath(fileName)
                Dim stream As Stream = New FileStream(filePath, FileMode.Open)
                Console.WriteLine(detector.Detect(stream))
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
            'ExEnd:ExtractEncodingByBOM
        End Sub

        ''' <summary>
        ''' Detects encoding from BOM is present or from the content if BOM is not present
        ''' </summary>
        ''' <param name="fileName"></param>
        Public Shared Sub ExtractEncodingByContentAndBOM(fileName As String)
            'ExStart:ExtractEncodingByContentAndBOM

            Try
                Dim detector As New EncodingDetector(Encoding.GetEncoding(1251))
                'get file actual path
                Dim filePath As [String] = Common.getFilePath(fileName)
                Dim stream As Stream = New FileStream(filePath, FileMode.Open)
                Console.WriteLine(detector.Detect(stream, True))
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
            'ExEnd:ExtractEncodingByContentAndBOM
        End Sub
    End Class

    Public Class logger
        ''' <summary>
        ''' Logs messages using NotificationReceiver 
        ''' </summary>
        ''' <param name="fileName"></param>
        Public Shared Sub LoggerWithManualExceptionHandling(fileName As String)
            'ExStart:LoggerWithManualExceptionHandling
            'get file actual path
            Dim filePath As [String] = Common.getFilePath(fileName)
            Dim receiver = New NotificationReceiver()
            Dim loadOptions As New LoadOptions()
            loadOptions.NotificationReceiver = receiver

            Try
                Using extractor = New CellsTextExtractor(filePath, loadOptions)
                    Console.WriteLine(extractor.ExtractAll())
                End Using
            Catch ex As Exception
                receiver.ProcessMessage(NotificationMessage.CreateErrorMessage(ex.Message, ex))
            End Try
            'ExEnd:LoggerWithManualExceptionHandling
        End Sub

        ''' <summary>
        ''' Logs messages using NotificationReceiver 
        ''' </summary>
        ''' <param name="fileName"></param>
        Public Shared Sub LoggerWithExtractorFactory(fileName As String)
            'ExStart:LoggerWithExtractorFactory
            'get file actual path
            Dim filePath As [String] = Common.getFilePath(fileName)
            Dim receiverForFactory = New NotificationReceiver()
            Dim factory = New ExtractorFactory(Nothing, Nothing, Nothing, receiverForFactory)

            Dim receiver = New NotificationReceiver()
            Dim loadOptions As New LoadOptions()
            loadOptions.NotificationReceiver = receiver

            Using extractor = New CellsTextExtractor(filePath, loadOptions)
                Console.WriteLine(extractor.ExtractAll())
            End Using
            'ExEnd:LoggerWithExtractorFactory
        End Sub
    End Class



    'ExStart:SimpleLogger
    Private Class NotificationReceiver
        Implements INotificationReceiver
        Public Sub ProcessMessage(message As NotificationMessage) Implements INotificationReceiver.ProcessMessage
            Console.WriteLine(message.Description)   
        End Sub
    End Class
    'ExEnd:SimpleLogger
End Class

File: Examples/CSharp/GroupDocs.Text.Examples.CSharp/Tools.cs
using GroupDocs.Text;
using GroupDocs.Text.Detectors.Encoding;
using GroupDocs.Text.Detectors.MediaType;
using GroupDocs.Text.Extractors.Text;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Text_for_.NET
{
    public class Tools
    {
        public class EncodingDetection
        {
            /// <summary>
            /// Detects encoding of a filestream when BOM is present
            /// </summary>
            /// <param name="fileName"></param>
            public static void ExtractEncodingByBOM(string fileName)
            {
                //ExStart:ExtractEncodingByBOM

                try
                {
                    EncodingDetector detector = new EncodingDetector(Encoding.GetEncoding(1251));
                    //get file actual path
                    String filePath = Common.GetFilePath(fileName);
                    Stream stream = new FileStream(filePath, FileMode.Open);
                    Console.WriteLine(detector.Detect(stream));
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex.Message);
                }
                //ExEnd:ExtractEncodingByBOM
            }

            /// <summary>
            /// Detects encoding from BOM is present or from the content if BOM is not present
            /// </summary>
            /// <param name="fileName"></param>
            public static void ExtractEncodingByContentAndBOM(string fileName)
            {
                //ExStart:ExtractEncodingByContentAndBOM

                try
                {
                    EncodingDetector detector = new EncodingDetector(Encoding.GetEncoding(1251));
                    //get file actual path
                    String filePath = Common.GetFilePath(fileName);
                    Stream stream = new FileStream(filePath, FileMode.Open);
                    Console.WriteLine(detector.Detect(stream, true));
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex.Message);
                }
                //ExEnd:ExtractEncodingByContentAndBOM
            }
        }

        public class logger
        {
            /// <summary>
            /// Logs messages using NotificationReceiver 
            /// </summary>
            /// <param name="fileName"></param>
            public static void LoggerWithManualExceptionHandling(string fileName)
            {
                //ExStart:LoggerWithManualExceptionHandling
                //get file actual path
                String filePath = Common.GetFilePath(fileName);
                var receiver = new NotificationReceiver();
                LoadOptions loadOptions = new LoadOptions();
                loadOptions.NotificationReceiver = receiver;

                try
                {
                    using (var extractor = new CellsTextExtractor(filePath, loadOptions))
                    {
                        Console.WriteLine(extractor.ExtractAll());
                    }
                }
                catch (Exception ex)
                {
                    receiver.ProcessMessage(NotificationMessage.CreateErrorMessage(ex.Message, ex));
                }
                //ExEnd:LoggerWithManualExceptionHandling
            }

            /// <summary>
            /// Logs messages using NotificationReceiver 
            /// </summary>
            /// <param name="fileName"></param>
            public static void LoggerWithExtractorFactory(string fileName)
            {
                //ExStart:LoggerWithExtractorFactory
                //get file actual path
                String filePath = Common.GetFilePath(fileName);
                var receiverForFactory = new NotificationReceiver();
                var factory = new ExtractorFactory(null, null, null, receiverForFactory);

                var receiver = new NotificationReceiver();
                LoadOptions loadOptions = new LoadOptions();
                loadOptions.NotificationReceiver = receiver;

                using (var extractor = new CellsTextExtractor(filePath, loadOptions))
                {
                    Console.WriteLine(extractor.ExtractAll());
                }
                //ExEnd:LoggerWithExtractorFactory
            }
        }



        //ExStart:SimpleLogger
        class NotificationReceiver : INotificationReceiver
        {
            public void ProcessMessage(NotificationMessage message)
            {
                Console.WriteLine(message.Description);
            }
        }
        //ExEnd:SimpleLogger

     
    }

}
